src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -90,7 +90,7 @@ object BitPat {
     */
   def apply(x: UInt): BitPat = {
     val len = if (x.isWidthKnown) x.getWidth else 0
-    apply("b" + x.litValue.toString(2).reverse.padTo(len, "0").reverse.mkString)
+    apply("b" + x.litOption.getOrElse(throw new ChiselException(s"$x is not a literal, BitPat.apply(x: UInt) only accept literal")).toString(2).reverse.padTo(len, "0").reverse.mkString)
```

Comment:
```suggestion
    x.litOption match {
      case Some(lit) => apply("b" + lit.toString(2).reverse.padTo(len, "0").reverse.mkString)
      case None => throwException(s"$x is not a literal, BitPat.apply(x: UInt) only accepts literals")
    }
```
Good to have a more clear error message, the line is just a bit long with it in a single line 🙂